### DISCUSSION TITLE
Manual/DynamicResolution

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#603] [Notes]
# Manual/DynamicResolution

https://docs.unity3d.com/Manual/DynamicResolution.html

**Comment by joonjoonjoon** [comment]
// Estimate the next frame time and update the resolution scale if necessary.

this comment is wrong. That function only calculates gpu and cpu frame times and puts them in a text field.
It does not update the resolution, or even attempt to.

Secondly using Screen.currentResolution is wrong, as it gives the resolution of the monitor, not the resolution of the game view or build window.
Thirdly, FrameTimingManager doesn't return any values for me on macOS, even though it is enabled in Player Settings.

**Comment by joonjoonjoon** [comment]
A seemingly better version of the script.
A Unity engineer also told me that FrameTimingManager does not work for some Unity versions before 2022.1, though was not able to be more specific. 
https://github.com/Unity-Technologies/DynamicResolutionSample/blob/main/DynamicResolution.cs
